src/GafferOSL/OSLObject.cpp

```diff
@@ -86,6 +87,41 @@ CompoundDataPtr prepareShadingPoints( const Primitive *primitive, const ShadingE
 		}
 	}
 
+	if( gafferAttributes )
+	{
+		for( const auto &i : gafferAttributes->members() )
+		{
+			if( shadingEngine->needsAttribute( i.first ) )
+			{
+				if( shadingPoints->writable().find( i.first ) == shadingPoints->writable().end() )
+				{
+					const IECore::Data* data = IECore::runTimeCast< IECore::Data >( i.second.get() );
+
+					// We currently don't support array attributes
+					// Using OpenImageIOAlgo to check if it's an array feels a bit weird, but it
+					// seems important to exactly match the logic of GafferOSL::ShadingEngine
+					if( data && !IECoreImage::OpenImageIOAlgo::DataView( data ).type.arraylen )
+					{
+						const IECore::BoolData* boolData = IECore::runTimeCast< const IECore::BoolData >( data );
+						if( boolData )
+						{
+							shadingPoints->writable()[i.first] = new IECore::IntData( boolData->readable() );
+						}
+						else
+						{
+							// Const cast is safe because the resulting dict is const
+							shadingPoints->writable()[i.first] = const_cast< IECore::Data* >( data );
+						}
+					}
+					else
+					{
+						throw IECore::Exception( "Invalid attribute for OSLObject : " + std::string( i.first ) + " of type " + i.second->typeName() );
```

Comment:
I'm a bit concerned about this, because `ShadingEngine::m_unknownAttributesNeeded` can mean that `ShadingEngine::attributesNeeded()` returns `true` for everything. That would make OSLObject unusable in most scenarios because it would be throwing here all the time due to ShaderNetworks and array data. It's not ideal, but I think the best we can do is silently carry on.